src/accelerate/utils/other.py

```diff
@@ -237,15 +242,17 @@ def load(f, map_location=None, **kwargs):
         **kwargs:
             Additional keyword arguments to pass to `torch.load()`.
     """
-    if is_weights_only_available():
-        if "weights_only" not in kwargs:
-            kwargs["weights_only"] = True
-        torch.serialization.add_safe_globals(TORCH_SAFE_GLOBALS)
-    else:
-        kwargs.pop("weights_only", None)
-    loaded_obj = torch.load(f, map_location=map_location, **kwargs)
-    if is_weights_only_available():
-        torch.serialization.clear_safe_globals()
+    try:
+        if is_weights_only_available():
+            if "weights_only" not in kwargs:
+                kwargs["weights_only"] = True
+            torch.serialization.add_safe_globals(TORCH_SAFE_GLOBALS)
+        else:
+            kwargs.pop("weights_only", None)
+        loaded_obj = torch.load(f, map_location=map_location, **kwargs)
+    finally:
+        if is_weights_only_available():
+            torch.serialization.clear_safe_globals()
```

Comment:
I wonder if this could happen: A user adds some safe globals, then uses accelerate, then wants to manually load something but now the safe globals are cleared. Maybe an edge case, but it would probably be better to restore the previous status quo instead of clearing completely.

Reply:
Yea I agree that this is a valid case as some libraries are adding to add_safe_globals on import (e.g. torchao, float8_experimental), perhaps we should gate on version >=2.5 and use the `safe_globals` ctx

(an alternative for now is you could technically create your own version of a context manager using a combination of `get/clear/add_{safe_globals}`, which are all available in 2.4)

Again apologies for the churn here

Reply:
I like the idea of backporting the context manager. Do you have standalone code to implement it? Probably something like:

```python
@contextmanager
def torch_safe_globals(types):
    # TODO check if torch has safe_globals, if yes use that, else:
    old_safe_globals = torch.serialization.get_safe_globals()
    torch.serialization.add_safe_globals(types)
    try:
        yield
    finally:
        torch.serialization.clear_safe_globals()
        torch.serialization.add_safe_globals(old_safe_globals)
```